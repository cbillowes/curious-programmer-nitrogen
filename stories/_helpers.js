import React from 'react'
import Layout from '../src/components/layout'
import { SmallSpacer } from '../src/components/spacer'

const Page = ({ className, children }) => {
  return (
    <Layout
      className={className}
    >
      <SmallSpacer />
      {children}
    </Layout>
  )
}

export default Page