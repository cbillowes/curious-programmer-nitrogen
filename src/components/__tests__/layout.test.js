import React from 'react'
import Layout from '../layout'
import { getSnapshot } from './_helpers'

describe(`Layout`, () => {
  it(`should render page`, () => {
    const layout = (
      <Layout>
        Hello World.
      </Layout>
    )
    const tree = getSnapshot(layout)
    expect(tree).toMatchSnapshot()
  })

  it(`should render reading pane`, () => {
    const layout = (
      <Layout
        reading={true}
      >
        Hello world in a reading page.
      </Layout>
    )
    const tree = getSnapshot(layout)
    expect(tree).toMatchSnapshot()
  })

  it(`should not render the footer`, () => {
    const layout = (
      <Layout
        footer="hidden"
      >
        Hello world without a footer.
      </Layout>
    )
    const tree = getSnapshot(layout)
    expect(tree).toMatchSnapshot()
  })

  it(`should render custom class`, () => {
    const layout = (
      <Layout
        className="the-answer-to-the-life-the-universe-and-everything"
      >
        Hello world in a custom class.
      </Layout>
    )
    const tree = getSnapshot(layout)
    expect(tree).toMatchSnapshot()
  })
})
