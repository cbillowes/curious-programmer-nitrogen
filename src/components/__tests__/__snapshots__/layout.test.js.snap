// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should not render the footer 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="wrapper"
    >
      <div
        className="logo"
      >
         
        <a
          className="full"
          href="/"
          title="Curious Programmer"
        >
          <strong>
            {
          </strong>
           :curious 
          <strong>
            "programmer"
          </strong>
           
          <strong>
            }
          </strong>
        </a>
         
         
        <a
          className="partial"
          href="/"
          title="Curious Programmer"
        >
          <strong>
            { }
          </strong>
        </a>
         
      </div>
      <button
        className="menu-button"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-ellipsis-v fa-w-6 "
          data-icon="ellipsis-v"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 192 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <nav
        className="navigation closed"
      >
        <div
          className="menu"
        >
           
          <a
            className="menu-item"
            href="/blog/"
          >
            Blog
          </a>
           
           
          <a
            className="menu-item"
            href="/about/"
          >
            About
          </a>
           
           
          <a
            className="menu-item"
            href="/tags/"
          >
            Tags
          </a>
           
           
          <a
            className="menu-item"
            href="/credits/"
          >
            Credits & Resources
          </a>
           
           
          <a
            className="menu-item"
            href="/license/"
          >
            License
          </a>
           
           
          <a
            className="menu-item"
            href="https://www.iubenda.com/privacy-policy/7895922"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            Privacy Policy
          </a>
           
           
          <a
            className="menu-item"
            href="https://github.com/cbillowes"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
           
           
          <a
            className="menu-item"
            href="https://www.linkedin.com/in/cbouwer/"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            LinkedIn
          </a>
           
        </div>
      </nav>
    </div>
  </header>
  <section
    className="container  undefined"
  >
    Hello world without a footer.
  </section>
</div>
`;

exports[`Layout should render custom class 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="wrapper"
    >
      <div
        className="logo"
      >
         
        <a
          className="full"
          href="/"
          title="Curious Programmer"
        >
          <strong>
            {
          </strong>
           :curious 
          <strong>
            "programmer"
          </strong>
           
          <strong>
            }
          </strong>
        </a>
         
         
        <a
          className="partial"
          href="/"
          title="Curious Programmer"
        >
          <strong>
            { }
          </strong>
        </a>
         
      </div>
      <button
        className="menu-button"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-ellipsis-v fa-w-6 "
          data-icon="ellipsis-v"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 192 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <nav
        className="navigation closed"
      >
        <div
          className="menu"
        >
           
          <a
            className="menu-item"
            href="/blog/"
          >
            Blog
          </a>
           
           
          <a
            className="menu-item"
            href="/about/"
          >
            About
          </a>
           
           
          <a
            className="menu-item"
            href="/tags/"
          >
            Tags
          </a>
           
           
          <a
            className="menu-item"
            href="/credits/"
          >
            Credits & Resources
          </a>
           
           
          <a
            className="menu-item"
            href="/license/"
          >
            License
          </a>
           
           
          <a
            className="menu-item"
            href="https://www.iubenda.com/privacy-policy/7895922"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            Privacy Policy
          </a>
           
           
          <a
            className="menu-item"
            href="https://github.com/cbillowes"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
           
           
          <a
            className="menu-item"
            href="https://www.linkedin.com/in/cbouwer/"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            LinkedIn
          </a>
           
        </div>
      </nav>
    </div>
  </header>
  <section
    className="container  the-answer-to-the-life-the-universe-and-everything"
  >
    Hello world in a custom class.
  </section>
</div>
`;

exports[`Layout should render page 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="wrapper"
    >
      <div
        className="logo"
      >
         
        <a
          className="full"
          href="/"
          title="Curious Programmer"
        >
          <strong>
            {
          </strong>
           :curious 
          <strong>
            "programmer"
          </strong>
           
          <strong>
            }
          </strong>
        </a>
         
         
        <a
          className="partial"
          href="/"
          title="Curious Programmer"
        >
          <strong>
            { }
          </strong>
        </a>
         
      </div>
      <button
        className="menu-button"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-ellipsis-v fa-w-6 "
          data-icon="ellipsis-v"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 192 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <nav
        className="navigation closed"
      >
        <div
          className="menu"
        >
           
          <a
            className="menu-item"
            href="/blog/"
          >
            Blog
          </a>
           
           
          <a
            className="menu-item"
            href="/about/"
          >
            About
          </a>
           
           
          <a
            className="menu-item"
            href="/tags/"
          >
            Tags
          </a>
           
           
          <a
            className="menu-item"
            href="/credits/"
          >
            Credits & Resources
          </a>
           
           
          <a
            className="menu-item"
            href="/license/"
          >
            License
          </a>
           
           
          <a
            className="menu-item"
            href="https://www.iubenda.com/privacy-policy/7895922"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            Privacy Policy
          </a>
           
           
          <a
            className="menu-item"
            href="https://github.com/cbillowes"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
           
           
          <a
            className="menu-item"
            href="https://www.linkedin.com/in/cbouwer/"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            LinkedIn
          </a>
           
        </div>
      </nav>
    </div>
  </header>
  <section
    className="container  undefined"
  >
    Hello World.
  </section>
</div>
`;

exports[`Layout should render reading pane 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="wrapper"
    >
      <div
        className="logo"
      >
         
        <a
          className="full"
          href="/"
          title="Curious Programmer"
        >
          <strong>
            {
          </strong>
           :curious 
          <strong>
            "programmer"
          </strong>
           
          <strong>
            }
          </strong>
        </a>
         
         
        <a
          className="partial"
          href="/"
          title="Curious Programmer"
        >
          <strong>
            { }
          </strong>
        </a>
         
      </div>
      <button
        className="menu-button"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-ellipsis-v fa-w-6 "
          data-icon="ellipsis-v"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 192 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <nav
        className="navigation closed"
      >
        <div
          className="menu"
        >
           
          <a
            className="menu-item"
            href="/blog/"
          >
            Blog
          </a>
           
           
          <a
            className="menu-item"
            href="/about/"
          >
            About
          </a>
           
           
          <a
            className="menu-item"
            href="/tags/"
          >
            Tags
          </a>
           
           
          <a
            className="menu-item"
            href="/credits/"
          >
            Credits & Resources
          </a>
           
           
          <a
            className="menu-item"
            href="/license/"
          >
            License
          </a>
           
           
          <a
            className="menu-item"
            href="https://www.iubenda.com/privacy-policy/7895922"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            Privacy Policy
          </a>
           
           
          <a
            className="menu-item"
            href="https://github.com/cbillowes"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            GitHub
          </a>
           
           
          <a
            className="menu-item"
            href="https://www.linkedin.com/in/cbouwer/"
            rel="nofollow noopener noreferrer"
            target="_blank"
          >
            LinkedIn
          </a>
           
        </div>
      </nav>
    </div>
  </header>
  <section
    className="container reading undefined"
  >
    Hello world in a reading page.
  </section>
</div>
`;
